### PR TITLE
Return AsyncIterable from execute

### PR DESCRIPTION
**Pull request by robrichard**

### CODE REVIEW

File: src/execution/dispatcher.js [hunk 1]
```diff
@@ -3,13 +3,48 @@
...
 import { type PromiseOrValue } from '../jsutils/PromiseOrValue';
+import { GraphQLError } from '../error/GraphQLError';
 import isPromise from '../jsutils/isPromise';
 
-import { GraphQLError } from '../error/GraphQLError';
+/**
+ * The result of GraphQL execution.
+ *
+ *   - `errors` is included when any errors occurred as a non-empty array.
+ *   - `data` is the result of a successful execution of the query.
```
Comment (lilianammmatos) [inline src/execution/dispatcher.js:15]: add `hasNext` to comment

File: src/execution/dispatcher.js [hunk 2]
```diff
@@ -3,13 +3,48 @@
...
+  data?: ObjMap<mixed> | null,
+  hasNext?: boolean,
+|};
+
+/**
+ * The result of an asynchronous GraphQL patch.
+ *
+ *   - `errors` is included when any errors occurred as a non-empty array.
+ *   - `data` is the result of the additional asynchronous data.
+ *   - `path` is the location of data .
```
Comment (lilianammmatos) [inline src/execution/dispatcher.js]: extra space here before `.`

File: src/execution/dispatcher.js [hunk 3]
```diff
@@ -3,13 +3,48 @@
...
+ *   - `label` is the label provided to @defer or @stream.
+ */
+export type ExecutionPatchResult = {|
+  errors?: $ReadOnlyArray<GraphQLError>,
+  data?: ObjMap<mixed> | mixed | null,
+  path: $ReadOnlyArray<string | number>,
+  label: string,
+  hasNext?: boolean,
```
Comment (lilianammmatos) [inline src/execution/dispatcher.js:38]: add `hasNext` to comment

File: src/execution/dispatcher.js [hunk 4]
```diff
@@ -23,33 +58,38 @@ export class Dispatcher {
...
     path: Path | void,
-    fn: () => PromiseOrValue<mixed>,
+    fn: () => PromiseOrValue<ObjMap<mixed> | mixed>,
     errors: Array<GraphQLError>,
   ) {
     this._patches.push(
-      this.execute(fn).then((data) => ({
-        value: {
+      this.execute(fn).then((data) => {
+        const value: $Shape<ExecutionPatchResult> = {
```
Comment (lilianammmatos) [inline src/execution/dispatcher.js:75]: Can we add the `hasNext` here that's always `true` to remove the need for `$Shape`?

Comment (robrichard) [reply src/execution/dispatcher.js:75]: it seems like its the errors spread that is confusing flow, not the missing `hasNext`

**Review by lilianammmatos — APPROVED** [review]

**Pull request merged**